components/radiobuttons/index.ts

```diff
@@ -0,0 +1,62 @@
+import m from "mithril";
+
+
+export type OptionValue = {
+    title?: string, 
+    value: string,
+    icon?: string,
+    disabled?:boolean,
+    id?: string
+}
+
+export type Attrs = {    
+    title: string,
+    value?: string, 
+    options: Array<OptionValue>
+    disabled?: boolean, 
+    onchange?: (event) => void;
+    onblur?: (event) => void;
+    onfocus?: (event) => void;
+}
+
+export type State = {
+    $container: HTMLElement;
+}
+
+export default {
+    view(vnode) {
+        const { value, options, disabled } = vnode.attrs;
+        return m(".mmf-radio-btn-container", {
+                disabled,
+                oncreate: _vnode => (this.$container = _vnode.dom as HTMLElement),
+            },
+            options.map((option: OptionValue) => {
+                const icon = option.icon ? 
+                    m("span.mmf-icon", {  
+                        value: option.value,
+                    }, option.icon) 
+                    : undefined;
+
+                return m(".mmf-radio-btn",
+                    {
+                        class: `${option.value === value ? "selected" : ""}`,
+                        value: option.value,
+                        disabled: disabled ? disabled : option.disabled || false,
+                        onclick: event => {
+                            event.target.focus();
+                            if ( disabled || option.disabled) return;
+                            Array.from(this.$container?.children).forEach(el => el.classList.remove("selected"));
+                            event.target?.parentNode?.classList?.add("selected");
```

Comment:
grabbing through dom is prone to errors. We could store our required dom-elements in oncreate-lifecycle hook